A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,641 @@
+// pass the dial option returned from this function as a dial option to
+// grpc.Dial().
+//
+// Using this option will always lead to instrumentation, however in order to
+// use the data a SpanExporter must be registered with the TraceProvider in the
+// TraceOption. Client side has retries, so a Unary and Streaming Interceptor are
+// registered to handle per RPC traces, and a Stats Handler is registered to handle
+// per RPC attempt trace. These three components registered work together in
+// conjunction, and do not work standalone.
+func DialOption(to TraceOptions) grpc.DialOption {}
+
+// ServerOption returns a server option which enables OpenTelemetry
+// instrumentation code for a grpc.Server.
+//
+// Server applications interested in instrumenting their grpc.Server should pass
+// the server option returned from this function as an argument to
+// grpc.NewServer().
+//
+// Using this option will always lead to instrumentation, however in order to
+// use the data a SpanExporter must be registered with the TraceProvider option.
+// Server side does not have retries, so a registered Stats Handler is the only
+// option that is returned.
+func ServerOption(to TraceOptions) grpc.ServerOption {}
+```
+
+## Tracing Information
+RPCs on the client side may undergo retry attempts, whereas on the server side, 
+they do not. gRPC records both per-call tracing details (on the parent span) 
+and per-attempt tracing details (on the attempt span) on the client side. 
+On the server side, there is only per-call traces.  With the new OpenTelemetry 
+plugin we will produce the following tracing information during an RPC lifecycle:
+
+At the client, on parent span:
+* If the RPC experienced name resolution delay, add an Event at the start of the
+  call with the name "Delayed name resolution complete" upon completion of the
+  name resolution process.
+* When the call is closed, set RPC status and end the parent span. gRPC status "OK" 
+  is recorded with status "OK", while other gRPC statuses are marked as "ERROR". 
+  Non-"OK" statuses include their code as a description, the same below.
+
+On attempt span:
+* When span is created, set the attribute with key `previous-rpc-attempts` and an 
+  integer value representing the count of previous attempts made for the RPC.
+* When span is created, set the attribute with key `transparent-retry` and a 
+  boolean value indicating whether the stream is undergoing a transparent retry.
+* If the RPC experienced load balancer pick delay, add an Event with the name 
+  "Delayed LB pick complete" upon creation of the stream on the transport. 
+* When the application sends an outbound message to the transport, add Event(s)
+  (it depends on implementation whether there is a single event or two separate 
+  events for compressed/uncompressed message sizes, the same below) 
+  with name "Outbound message sent" and the following attributes:
+  * key `message.event.type` with string value "SENT".
```

Comment:
is this attribute "message.event.type" useful? The message already says sent?

Reply:
Fair enough, removed.